app/routers/users.py

```diff
@@ -215,10 +228,27 @@ async def delete_user(id: str):  # pragma: no cover
 @router.get(
     "/{id}/accounts",
     dependencies=[Depends(authentication_required)],
-    response_model=list[AccountUserRead],
+    response_model=LimitOffsetPage[AccountUserRead],
 )
-async def get_user_accounts(id: str):  # pragma: no cover
-    pass  # not yet implemented
+async def get_user_accounts(
+    user: Annotated[User, Depends(fetch_user_or_404)],
+    accountuser_repo: AccountUserRepository,
+    auth_ctx: CurrentAuthContext,
+):
+    extra_conditions: list[ColumnExpressionArgument] = [
+        AccountUser.user == user,
+    ]
+    if auth_ctx is not None and auth_ctx.account.type == AccountType.AFFILIATE:
+        extra_conditions += [
+            AccountUser.account.has(Account.status != AccountStatus.DELETED),
```

Comment:
This condition is not needed.
For the moment there is no possibility to delete an account.
In the future, I expect that, deleting an account cascade on its AccountUser

Reply:
In any case if we return a page of Accounts, the query will be done on the Account table joining the AccountUser